src/common/SIRF.py

```diff
@@ -155,6 +155,31 @@ def get_uniform_copy(self, value=1.0):
         y.fill(value)
         return y
 
+    def fill(self, value=0):
+        return
+
+    def allocate(self, value=0, **kwargs):
+        """Allocates a copy of self and fills with values
+
+        CIL/SIRF compatibility
+        """
+        out = self.clone()
```

Comment:
can we call `get_uniform_copy` here? Not all derived classes might have it of course, in which case I'd suggest to add it. It could potentially save a fair amount of time copying the data (sadly expensive for PET...).

If we can, it'd make sense to do
```python
if value in ['random', 'random_int']:
  out =  self.get_uniform_copy()
 ...
  return out
else:
  if value is None:
    value = 0
  return out.get_uniform_copy(value)
```

Reply:
a good idea, thanks!